### PR TITLE
Feature: Data augmentation

### PR DESCRIPTION
**Pull request by hlfshell**
The dataloader is modified to create data augmentation on our dataset. It now performs the following operations:

- 50% chance of horizontal flip
- Contrast is randomly adjusted
- Brightness is lowered or raised
- Gaussian blur is added
- Salt noise randomly added.
Trainer will not add data augmentation to the validation.

### CODE REVIEW

**Comment by Rhodwarrior** [comment]
If im reading corectly, we chance mirror, then apply all of the filters to one degree or other and pass back.  Couldnt we get more augmentation by definitiely mirroring, then apply brightness to both mirrored and non, apply contrast to both etc.  and pass back label files (mirrored and regular) and all of the corresponding image files( not mirrored, mirrored, nm with constrast adj, mirror with contrast, nm with brightness, mirror with brightness etc) iguess I'm just saying with one augmentation call we could get 8-10 augmented images back.

**Comment by hlfshell** [comment]
> Couldnt we get more augmentation by definitiely mirroring, then apply brightness to both mirrored and non, apply contrast to both etc. and pass back label files (mirrored and regular) and all of the corresponding image files( not mirrored, mirrored, nm with constrast adj, mirror with contrast, nm with brightness, mirror with brightness etc) iguess I'm just saying with one augmentation call we could get 8-10 augmented images back.

Because it's slightly harder. Batch size is defined as the limits of our total system memory, so it's weirder that we specify batch size of `M` but because of data augmentation we end up with a larger batch size of `M` (`N>M`). We'll have to figure out how to pass the difference down to other training pieces that expect that batch number to be known up front, etc.

It's best to do a bunch of randomized changes so you never quite get the same image twice.

File: unet/dataset.py [hunk 1]
```diff
@@ -43,18 +90,55 @@ def __getitem__(self, start):
...
+        #     # interpolation that uses nearest neighbor and avoids any kind of averaging
+        #     # since that would be meaningless in a labels approach. This is what order=0
+        #     # is doing below.
+        #     # https://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.resize
+        #     labels = resize(labels, self.img_size, order=0)
+            
+        #     # labels is (x, y) and we want labels to be (x, y, 1) in shape
+        #     labels = np.expand_dims(labels, 2)
+
+        #     Y[index] = labels
```
Comment (zekeflaton) [inline unet/dataset.py]: any reason to keep this if it's commented out?

**Review by zekeflaton — DISMISSED** [review]
LGTM

File: unet/dataset.py (hunk 1 above)
Comment (hlfshell) [reply unet/dataset.py]: Yeah that's a mistake. Cleared it up.

**Review by zekeflaton — APPROVED** [review]
